### DISCUSSION TITLE
Serverless Function called by CRON is "request" type

### DISCUSSION THREAD

**Discussion by Jamesllllllllll** [vercel/community#4067] [Help] [answered]
### Summary

I've scheduled a CRON job to run a Serverless Function. However, when the CRON job runs, the function's **type** is "request", and the it does not execute the work. It returns a 200 status.

![Screenshot 2023-09-13 233648](https://github.com/vercel/community/assets/125431058/37dcfca0-d29a-4158-a270-c1ccea24d17c)

My `vercel.json`:

```js
{
  "crons": [
    {
      "path": "/api/weeklyCron",
      "schedule": "*/5 * * * *"
    }
  ]
}
```

I have added other Serverless Functions to "crons" and they are called correctly:

> NOTE: The below function was called for testing purposes. It requires query params, which were not included for the test, which causes the error

![Screenshot 2023-09-13 233733](https://github.com/vercel/community/assets/125431058/65c1e8f6-b47c-4eb2-a630-1b16bc693a0a)

I'm using the **App** router, so here is my `/src/app/api/weeklyCron/route.js`:

```js
import { NextResponse } from 'next/server';

const mockDailyUsers = [
  {
    id: 1,
    username: 'james',
    email: /* omitted */`,
  },
];

export async function GET(req, res) {
  const baseUrl =
    process.env.NODE_ENV === 'production'
      ? `https://beacons-link-checker.vercel.app`
      : `http://localhost:3000`;

  const checkedUsers = [];

  for (let i = 0; i < mockDailyUsers.length; i++) {
    const username = mockDailyUsers[i].username;
    const email = mockDailyUsers[i].email;
    const url = `${baseUrl}/api/checkWeekly?user=${username}&email=${email}`;
    console.log(`CRON job checking: ${username}`);
    const check = await fetch(url, { method: 'GET' });
    console.log(check);
    checkedUsers.push(username);
  }

  return NextResponse.json({ data: checkedUsers }, { status: 200 });
}
```

### Steps to Reproduce

To replicate, you would need to clone and deploy the repo - I wouldn't expect that! I've searched for answers about serverless functions being just "requests", to no avail.

The API route will return the list of added users (I put it there to test) - so I'm not sure what else I can do to make this work!

Thank you in advance for any help.

**Comment by amyegan** [comment, marked as answer]
Does it work if you add `export const dynamic = 'force-dynamic'` [route segment config](https://nextjs.org/docs/app/building-your-application/routing/route-handlers#segment-config-options)? This sounds similar to https://github.com/vercel/community/discussions/3903#discussioncomment-6926884

**Reply by Jamesllllllllll** [reply]
Oh dear, this is exactly the answer! Thank you, @amyegan - I appreciate your help. Apologies for the duplicate issue.
